pkg/kubectl/cmd/portforward/portforward.go

```diff
@@ -108,15 +108,9 @@ func NewCmdPortForward(f cmdutil.Factory, streams genericclioptions.IOStreams) *
 		Long:                  portforwardLong,
 		Example:               portforwardExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			if err := opts.Complete(f, cmd, args); err != nil {
-				cmdutil.CheckErr(err)
-			}
-			if err := opts.Validate(); err != nil {
-				cmdutil.CheckErr(cmdutil.UsageErrorf(cmd, "%v", err.Error()))
-			}
-			if err := opts.RunPortForward(); err != nil {
-				cmdutil.CheckErr(err)
-			}
+			cmdutil.CheckErr(opts.Complete(f, cmd, args))
+			cmdutil.CheckErr(opts.Validate())
```

Comment:
Previously in case of error in validation we were calling the `cmdutil.UsageErrorf` function which adds `See '%s -h' for help and examples.` to the end of the error message. I think, we still need to call it. Am I wrong?

Reply:
I  also think, help information printed is more kind.But i found some errors print like this while  others not.
For example
pkg/kubectl/cmd/run/run.go
```go
	if len(args) == 0 || o.ArgsLenAtDash == 0 {
		return cmdutil.UsageErrorf(cmd, "NAME is required for run")
	}

	timeout, err := cmdutil.GetPodRunningTimeoutFlag(cmd)
	if err != nil {
		return cmdutil.UsageErrorf(cmd, "%v", err)
	}

	// validate image name
	imageName := o.Image
	if imageName == "" {
		return fmt.Errorf("--image is required")
	}
	validImageRef := reference.ReferenceRegexp.MatchString(imageName)
	if !validImageRef {
		return fmt.Errorf("Invalid image name %q: %v", imageName, reference.ErrReferenceInvalidFormat)
	}

	if o.TTY && !o.Interactive {
		return cmdutil.UsageErrorf(cmd, "-i/--stdin is required for containers with -t/--tty=true")
	}
	replicas := cmdutil.GetFlagInt(cmd, "replicas")
	if o.Interactive && replicas != 1 {
		return cmdutil.UsageErrorf(cmd, "-i/--stdin requires that replicas is 1, found %d", replicas)
	}

```
pkg/kubectl/cmd/taint/taint.go
```go
	if len(o.resources) < 1 {
		return fmt.Errorf("one or more resources must be specified as <resource> <name>")
	}
	if len(taintArgs) < 1 {
		return fmt.Errorf("at least one taint update is required")
	}

	if o.taintsToAdd, o.taintsToRemove, err = parseTaints(taintArgs); err != nil {
		return cmdutil.UsageErrorf(cmd, err.Error())
	}
```
Is there any necessary to make these returns  follow the same pattern.
If necessary ,i will be very happy to check all commands and make returns which are usage related  return with "See '%s -h' for help and examples.'.

Reply:
> i will be very happy to check all commands

I think, would be great to use `cmdutil.UsageErrorf` when the error is related to invalid input. It probably worth double checking with [sig-cli in Slack/mailing list](https://github.com/kubernetes/community/tree/master/sig-cli#contact) before spending a lot of time going through all the commands.

Thanks for heads up!